src/Sulu/Bundle/DocumentManagerBundle/DependencyInjection/Compiler/DocumentFixturePass.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\DocumentManagerBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class DocumentFixturePass implements CompilerPassInterface
+{
+    const TAG_NAME = 'sulu.document_manager_fixture';
+
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->findTaggedServiceIds(self::TAG_NAME) as $id => $tags) {
+            $definition = $container->getDefinition($id);
+
+            if (is_subclass_of($definition->getClass(), ContainerAwareInterface::class)) {
+                @trigger_error(
```

Comment:
Do we still need that Compiler pass? Shouldn't that also be something we can safely delete now?

Reply:
No this does not work with symfony. Symfony does not autowire the ContainerAwareInterface to make the update as smooth as possible I would keep it.

Reply:
I was especially wondering, because the Fixture on our website implements that `ContainerAwareInterface`, and I still can't find this deprecation anywhere... Can you explain why that is? I assumed that this code is not called for service-container registered features because of that, which would make that class obsolete.

Reply:
The `call` for `setContainer` is not set without the CompilerPass:

```bash
bin/console debug:container App\DataFixtures\Document\DocumentFixture --show-arguments
```

If you disable the compilerpass and clear the cache you will then get:

```bash
Loading fixtures
 - loading "App\DataFixtures\Document\DocumentFixture"
Symfony\Component\Debug\Exception\FatalThrowableError^ {#2035
  -originalClassName: "Error"
  #message: "Call to a member function get() on null"
  #code: 0
  #file: "./src/DataFixtures/Document/DocumentFixture.php"
```

> I still can't find this deprecation anywhere

Maybe the deprecation isn't logged correctly because its a compilerpass, I will add also a deprecation to the command itself.

Reply:
@danrot deprecation message also added to the command